core/ipc/twin/common/src/main/java/org/opennms/core/ipc/twin/common/AbstractTwinSubscriber.java

```diff
@@ -0,0 +1,98 @@
+/*******************************************************************************
+ * This file is part of OpenNMS(R).
+ *
+ * Copyright (C) 2021 The OpenNMS Group, Inc.
+ * OpenNMS(R) is Copyright (C) 1999-2021 The OpenNMS Group, Inc.
+ *
+ * OpenNMS(R) is a registered trademark of The OpenNMS Group, Inc.
+ *
+ * OpenNMS(R) is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published
+ * by the Free Software Foundation, either version 3 of the License,
+ * or (at your option) any later version.
+ *
+ * OpenNMS(R) is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with OpenNMS(R).  If not, see:
+ *      http://www.gnu.org/licenses/
+ *
+ * For more information contact:
+ *     OpenNMS(R) Licensing <license@opennms.org>
+ *     http://www.opennms.org/
+ *     http://www.opennms.com/
+ *******************************************************************************/
+
+package org.opennms.core.ipc.twin.common;
+
+import java.io.Closeable;
+import java.io.IOException;
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.function.Consumer;
+
+import org.opennms.core.ipc.twin.api.TwinSubscriber;
+import org.opennms.distributed.core.api.MinionIdentity;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+
+public abstract class AbstractTwinSubscriber implements TwinSubscriber {
+
+    private final Map<String, Class<?>> classesByKey = new ConcurrentHashMap<>();
+    private final Map<String, Consumer<?>> consumerMap = new ConcurrentHashMap<>();
+    private final Map<Closeable, String> closableMap = new ConcurrentHashMap<>();
```

Comment:
Can we unify this by also having a `SessionImpl` in here.
Something like this:
```java
private class SessionImpl implements Closable {
    private final String key;
    private final Class<?> clazz;
    private final Consumer<?> consumer;

    public void accept(TwinResponseBean twinResponse) throws IOException {
        final var value = objectMapper.readValue(twinResponse.value, this.clazz);
        this.consumer.accpet(value);
    }

    public void close() {
        AbstractTwinSubscriber.this.sessions.remove(this.key);
    }
}
```

This can be used in a single `Map<String, Session>` whereas the `Map`s key is the session key.
`getObject` the could do a `computeIfAbsend` on this `Map` and return the object from the `Map`.